test/e2e/metrics_test.go

```diff
@@ -26,6 +26,10 @@ const (
 )
 
 func testMetrics(t *testing.T) {
+	testCtx, err := NewTestContext()
+	require.NoError(t, err)
+	require.NoError(t, testCtx.createNamespace(testCtx.workloadNamespace), "error creating test namespace")
```

Comment:
Why is this change being made? Is it because the metrics tests now run before the networking tests when run in CI? This will cause issues if the network tests are run independently. Instead the workload namespace creation/deletion can be handled in the common `TestWMCO` function.

Reply:
>Is it because the metrics tests now run before the networking tests when run in CI?

Yes

> Instead the workload namespace creation/deletion can be handled in the common TestWMCO function.

Moving the creation is not straightforward as `testCtx.createNamespace` would need to become `testCtx.EnsureNamespace`. Given how rarely we run the network tests independently I decided to put this off. But I can make the change as part of this PR.